lnwire/accept_channel.go

```diff
@@ -113,7 +118,15 @@ func (a *AcceptChannel) Encode(w io.Writer, pver uint32) error {
 		a.DelayedPaymentPoint,
 		a.HtlcPoint,
 		a.FirstCommitmentPoint,
-	)
+	); err != nil {
+		return err
+	}
+
+	if len(a.UpfrontShutdownScript) == 0 {
```

Comment:
The spec says 
```
if both nodes advertised the option_upfront_shutdown_script feature:
MUST include either a valid shutdown_scriptpubkey as required by shutdown scriptpubkey, or a zero-length shutdown_scriptpubkey.`
```

I read this as we must write a zero length field in this case?

Reply:
Checked this out and you're right about the 0 value 👍 

So proceeding with the following:
- Encode a 0 for empty shutdown address (fine for non upfront shutdown understanding lnd, will have to double check interop)
- Encode `len`/`addr` when given a non-0 delivery address
- Always try to decode with the EOF trick